stub-runner/stub-runner-spring/src/main/groovy/com/ofg/stub/StubDownloader.groovy

```diff
@@ -0,0 +1,157 @@
+package com.ofg.stub
+
+import com.ofg.stub.spring.ZipCategory
+import groovy.util.logging.Slf4j
+
+import static groovy.grape.Grape.addResolver
+import static groovy.grape.Grape.resolve
+import static groovy.io.FileType.FILES
+import static java.nio.file.Files.createTempDirectory
+
+@Slf4j
+class StubDownloader {
+
+    private static final String LATEST_MODULE = '*'
+    private static final String REPOSITORY_NAME = 'dependency-repository'
+    private static final String STUB_RUNNER_TEMP_DIR_PREFIX = 'stub-runner'
+
+    private DependencyResolver dependencyResolver
+
+    StubDownloader() {
+    }
+
+    StubDownloader(DependencyResolver dependencyResolver) {
+        this.dependencyResolver = dependencyResolver
+    }
+
+    File downloadAndUnpackStubJar(boolean skipLocalRepo, String stubRepositoryRoot, String stubsGroup, String
+            stubsModule) {
+        URI stubJarUri = findGrabbedStubJars(skipLocalRepo, stubRepositoryRoot, stubsGroup, stubsModule)
+        File unzippedStubsDir = unpackStubJarToATemporaryFolder(stubJarUri)
+        return unzippedStubsDir
+    }
+
+    private File unpackStubJarToATemporaryFolder(URI stubJarUri) {
+        File tmpDirWhereStubsWillBeUnzipped = createTempDirectory(STUB_RUNNER_TEMP_DIR_PREFIX).toFile()
+        tmpDirWhereStubsWillBeUnzipped.deleteOnExit()
+        log.debug("Unpacking stub from JAR [URI: ${stubJarUri}]")
+        use(ZipCategory) {
+            new File(stubJarUri).unzipTo(tmpDirWhereStubsWillBeUnzipped)
+        }
+        return tmpDirWhereStubsWillBeUnzipped
+    }
+
+    private URI findGrabbedStubJars(boolean skipLocalRepo, String stubRepositoryRoot, String stubsGroup, String stubsModule) {
+        Map depToGrab = [group: stubsGroup, module: stubsModule, version: LATEST_MODULE, transitive: false]
+        setDependencyResolver(skipLocalRepo)
+        return dependencyResolver.resolveDependency(stubRepositoryRoot, depToGrab)
+    }
+
+    private setDependencyResolver(boolean skipLocalRepo) {
+        if (this.dependencyResolver == null) {
+            this.dependencyResolver = buildResolver(skipLocalRepo)
+        }
+
+    }
+
+    private DependencyResolver buildResolver(boolean skipLocalRepo) {
+        skipLocalRepo ? new RemoteDependencyResolver() : new LocalFirstDependencyResolver()
+    }
+
+    /**
+     * Dependency resolver providing {@link URI} to remote dependencies.
+     */
+    @Slf4j
+    private class RemoteDependencyResolver extends DependencyResolver {
+
+        URI resolveDependency(String stubRepositoryRoot, Map depToGrab) {
+            try {
+                return doResolveRemoteDependency(stubRepositoryRoot, depToGrab)
+            }
+            catch (UnknownHostException e) {
```

Comment:
Why catch in the separate line? The following is more compact (and is usually used in Java/Groovy code):

```
            try {
                return doResolveRemoteDependency(stubRepositoryRoot, depToGrab)
            } catch (UnknownHostException e) {
                failureHandler(stubRepositoryRoot, "unknown host error -> ${e.message}", e)
            } catch (Exception e) {
                failureHandler(stubRepositoryRoot, "connection error -> ${e.message}", e)
            }
```